src/main/kotlin/assertk/assertions/inputstream.kt

```diff
@@ -0,0 +1,118 @@
+package assertk.assertions
+
+import assertk.Assert
+import assertk.assertions.support.expected
+import java.io.InputStream
+
+/** Asserts that the actual stream has the same content as the other stream.
+ *
+ * @param other whichs content is compared to the actual one
+ */
+fun Assert<InputStream>.hasSameContentAs(other: InputStream) {
+
+    var size = 0L
+
+    val actualBuffer = ByteArray(BUFFER_SIZE)
+    val otherBuffer = ByteArray(BUFFER_SIZE)
+
+    while (true) {
+        val actualRead = fillBuffer(actual, actualBuffer)
+        val otherRead = fillBuffer(other, otherBuffer)
+
+        if (actualRead == otherRead) {
+
+            if(actualRead==0) {
+                // the streams have the same size and are finished and have the same content
+                return
+            }
+            else {
+                val pos = compare(actualRead, actualBuffer, otherBuffer)
+                if(pos==null) {
+                    // the current buffers are equals and we can try the next block
+                    size += actualRead
+
+                } else {
+                    // the first difference is in position pos
+
+                    val actualSize = size + actualRead + consume(actual)
+                    val otherSize = size + otherRead + consume(other)
+
+                    expected("to have the same content, but actual stream differs at pos $size. Actual stream: value=0x${actualBuffer[pos].toHexString()}, size=$actualSize. Other stream: value=0x${otherBuffer[pos].toHexString()}, size=$otherSize")
```

Comment:
You can't assume that `expected` will terminate/throw an exception. If you are using something like `assertAll` it will collect all failures, so you need to break if you don't want to continue the loop.